src/body-builder.js

```diff
@@ -0,0 +1,197 @@
+import _ from 'lodash'
+import {boolMerge} from './utils'
+
+function bodyBuilder () {
+  return Object.assign(
+    { /* sort, from, build, etc. go here */ },
+    queryBuilder(),
+    filterBuilder(),
+    aggregationBuilder()
+  )
+}
+
+/**
+ * Generic builder for filter and query clauses
+ *
+ * @private
+ *
+ * @param  {string} field
+ * @param  {any}    value
+ * @param  {Object} opts
+ * @return {Object}       query clause component
+ */
+function buildClause (field, value, opts) {
```

Comment:
Yep. I was even thinking about giving `bodyBuilder`, `queryBuilder`, `aggregationBuilder` and `filterBuilder` each a separate file. Different `buildClause` for query and filter builders would be necessary anyways.